### ISSUE TITLE
`Concatenate` is not rejected when used in a simple type annotation

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6898]
The following should generate a type error, but it doesn't currently.

```python
def func2(x: Concatenate[int, P]) -> int:  # Type error
    ...
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.345, which I just published. It will also be included in a future release of pylance.
